core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -91,6 +97,64 @@ public DeterministicKey(ImmutableList<ChildNumber> childNumberPath,
         this.keyCrypter = checkNotNull(crypter);
     }
 
+    /**
+     * Return the fingerprint of this key's parent as an int value, or zero if this key is the
+     * root node of the key hierarchy.  Raise an exception if the arguments are inconsistent.
+     * This method exists to avoid code repetition in the constructors.
+     */
+    private int ascertainParentFingerprint(DeterministicKey parentKey, byte[] parentFingerprint)
```

Comment:
Oh,hmm, @devrandom do you know of any reason why fingerprints would be represented as a byte[4]? My preference would be for changing it as an int is a lot more efficient.